ilastik/workflows/neuralNetwork/_localWorkflow.py

```diff
@@ -56,6 +56,29 @@ def _createClassifierApplet(self):
         conn_str = self._launcher.start()
         srv_config = ServerConfig(id="auto", address=conn_str, devices=[Device(id="cpu", name="cpu", enabled=True)])
         connFactory = tiktorch.TiktorchConnectionFactory()
+        conn = connFactory.ensure_connection(srv_config)
+
+        devices = conn.get_devices()
+        preferred_cuda_device_id = runtime_cfg.preferred_cuda_device_id
+        device_ids = [dev[0] for dev in devices]
+        cuda_devices = tuple(d for d in device_ids if d.startswith("cuda"))
+
+        if preferred_cuda_device_id not in device_ids:
+            if preferred_cuda_device_id:
+                logger.warning(f"Could nor find preferred cuda device {preferred_cuda_device_id}")
+            try:
+                preferred_cuda_device_id = cuda_devices[0]
+            except IndexError:
+                preferred_cuda_device_id = "cpu"
+
+            logger.info(f"Using default device for Neural Network Workflow {preferred_cuda_device_id}")
+        else:
+            logger.info(f"Using specified device for Neural Netowrk Workflow {preferred_cuda_device_id}")
+
+        device_name = devices[device_ids.index(preferred_cuda_device_id)][1]
+
+        srv_config = srv_config.evolve(devices=[Device(id=preferred_cuda_device_id, name=device_name, enabled=True)])
```

Comment:
```suggestion
        device_id = runtime_cfg.preferred_cuda_device_id
        devices = {d[0]: d[1] for d in conn.get_devices()}

        if device_id in devices:
            selection_policy = "specified"
        else:
            if device_id:
                warnings.warn(f"Could not find preferred CUDA device {device_id}")
            selection_policy = "default"
            device_id = next((d for d in devices if d.startswith("cuda")), "cpu")

            logger.info(f"Using {selection_policy} device for Neural Network Workflow {device_id}")

        srv_config = srv_config.evolve(devices=[Device(id=device_id, name=devices[device_id], enabled=True)])
```

Reply:
looks really good. Will favor the "tested" current code for now, but you can maybe open a PR with this!